gnomad/assessment/sanity_checks.py

```diff
@@ -103,90 +117,850 @@ def make_filters_sanity_check_expr(
+    :param metrics: List of metrics to sum and compare to annotationed versions. Default is ["AC", "AN", "nhomalt"].
+    :return: Dictionary of sample sum field check expressions and display fields.
     """
-    if prefix != "":
-        prefix = f"{prefix}_"
+    t = t.rows() if isinstance(t, hl.MatrixTable) else t
 
-    label_combos = make_label_combos(label_groups)
-    combo_AC = [ht.info[f"{prefix}AC_{x}"] for x in label_combos]
-    combo_AN = [ht.info[f"{prefix}AN_{x}"] for x in label_combos]
-    combo_nhomalt = [ht.info[f"{prefix}nhomalt_{x}"] for x in label_combos]
+    # An empty string is passed to run this check on the entire callset but we do not want to add an delimiter to the field look ups for the empty string
+    if subset:
+        subset += delimiter
 
+    label_combos = make_label_combos(label_groups, label_delimiter=delimiter)
+    # Grab the adj group for checks as we do not retain the raw metric counts for all sample groups so we cannot check the sample sums
     group = label_groups.pop("group")[0]
-    alt_groups = "_".join(
+    # sum_group is a subset of group. When the metrics of sum_group are added, they should equal the total metric of group.
+    # e.g., if group = callset1, one sum_group could be callset1-pop
+    sum_group = delimiter.join(
         sorted(label_groups.keys(), key=lambda x: sort_order.index(x))
     )
+    info_fields = t.info.keys()
 
-    annot_dict = {
-        f"sum_AC_{group}_{alt_groups}": hl.sum(combo_AC),
-        f"sum_AN_{group}_{alt_groups}": hl.sum(combo_AN),
-        f"sum_nhomalt_{group}_{alt_groups}": hl.sum(combo_nhomalt),
-    }
+    # Loop through metrics to build dictionary of fields that make up 'group' and then store the metrics sums
+    annot_dict = {}
+    for metric in metrics:
+        sum_group_exprs = []
+        for label in label_combos:
+            if metric_first_label:
+                field = f"{metric}{delimiter}{subset}{label}"
+            else:
+                field = f"{subset}{metric}{delimiter}{label}"
 
-    ht = ht.annotate(**annot_dict)
+            if field in info_fields:
+                sum_group_exprs.append(t.info[field])
+            else:
+                logger.info("%s is not in table's info field", field)
 
-    for subfield in ["AC", "AN", "nhomalt"]:
-        if not subpop:
-            generic_field_check(
-                ht,
-                (
-                    ht.info[f"{prefix}{subfield}_{group}"]
-                    != ht[f"sum_{subfield}_{group}_{alt_groups}"]
-                ),
-                f"{prefix}{subfield}_{group} = sum({subfield}_{group}_{alt_groups})",
-                [
-                    f"info.{prefix}{subfield}_{group}",
-                    f"sum_{subfield}_{group}_{alt_groups}",
-                ],
-                verbose,
-            )
+        annot_dict[
+            f"sum{delimiter}{metric}{delimiter}{group}{delimiter}{sum_group}"
```

Comment:
Ah I see, the previous message was actually not what was really being checked because it didn't include the subset info, so now it is more accurate. As for the repeated displays, if there is now way around the repeat then I would prefer to keep it faster and have the repeat.

Reply:
Actually, I think there might be an extra indent on generic_field_check_loop(t, field_check_expr, verbose)

Reply:
You're right!